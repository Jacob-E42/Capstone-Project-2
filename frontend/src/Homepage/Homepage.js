import React, { useContext } from "react";
import { Button, Typography, Box } from "@mui/material";
import { Link } from "react-router-dom";
import UserContext from "../context_providers/UserContext";

const Homepage = () => {
	const { user } = useContext(UserContext);
	console.debug("Homepage", "user=", user);

	return (
		<Box
			sx={{
				height: "100vh", // Full view height
				m: 0,
				display: "flex",
				flexDirection: "column",
				justifyContent: "center",
				alignItems: "center",
				backgroundImage: "linear-gradient(120deg, #84fab0 0%, #8fd3f4 100%)" // Example gradient
			}}>
			<Box
				sx={{
					maxWidth: 500
				}}>
				<img
					src="/logos/inkling_homepage_logo.png" // Replace with your image path
					alt="Inkling Logo"
					style={{
						maxWidth: "100%", // Ensures the image is responsive
						maxHeight: "300px",
						height: "auto",
						marginBottom: "30px" // Adds some space between the image and the title
					}}
				/>
				{/* <Typography
					variant="h2"
					gutterBottom>
					Inkling
				</Typography> */}
				<Typography variant="h5">
					Capture Your Thoughts, <br />
					Discover Your Insights
				</Typography>
				<Box sx={{ "& > *": { m: 1.25 } }}>
					<Button
						variant="contained"
						color="secondary"
						component={Link}
						to="/signup">
						Sign Up
					</Button>
					<Button
						variant="contained"
						color="secondary"
						component={Link}
						to="/login">
						Log In
					</Button>
				</Box>
			</Box>
		</Box>
	);
};

export default Homepage;
