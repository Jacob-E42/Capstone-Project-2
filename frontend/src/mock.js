import React, { useCallback, useState } from "react";
import UserContext from "./context_providers/UserContext";
import AlertContext from "./context_providers/AlertContext";
import Request from "./api";
import ApiRequest from "./api";
import ApiContext from "./context_providers/ApiContext";
import JournalContext from "./context_providers/JournalContext";

const demoUser = {
	id: 11,
	firstName: "testfirst",
	lastName: "testlast",
	email: "test@test.net",
	interests: ["baking", "jetskiing"]
};

const demoJournal = {
	id: 42,
	userId: 11,
	title: "Baking adventure",
	entryText:
		"Yesterday I tried baking for the first time like I've always wanted to. It was a complete disaster, but at least it was fun.",
	entryDate: "2023-07-24",
	emotions: null
};

const UserProvider = ({ children }) => {
	return <UserContext.Provider value={{ user: demoUser }}>{children}</UserContext.Provider>;
};

const AnonUserProvider = ({ children }) => {
	const [currentUser, setCurrentUser] = useState(null);

	const login = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.login(formData);
				if (token) {
					setCurrentUser(demoUser);
					return { success: true };
				} else throw new Error("Log in token in missing");
			} catch (errors) {
				console.error("login failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	const signup = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.signup(formData);
				console.log(token);
				if (token) setCurrentUser(demoUser);
				else throw new Error("missing sign in token");
				return { success: true };
			} catch (errors) {
				console.error("signup failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	return <UserContext.Provider value={{ currentUser, login, signup }}>{children}</UserContext.Provider>;
};

const AlertProvider = ({ children }) => {
	const [msg, setMsg] = useState("");
	const [color, setColor] = useState("primary");

	return <AlertContext.Provider value={{ msg, setMsg, color, setColor }}>{children}</AlertContext.Provider>;
};

const ApiProvider = ({ children }) => {
	const [request, setRequest] = useState(new ApiRequest());
	request.setToken(
		`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTEsImVtYWlsIjoidGVzdEB0ZXN0Lm5ldCIsImlhdCI6MTY5MjI5MDkxN30.8e12Cw-TMts0FmZALH61ch03kUMlW5QoCi60dlfhyWY`
	);

	return <ApiContext.Provider value={{ api: request }}>{children}</ApiContext.Provider>;
};

const JournalProvider = ({ children }) => {
	return <JournalContext.Provider value={{ journal: demoJournal }}>{children}</JournalContext.Provider>;
};

export { UserProvider, AnonUserProvider, AlertProvider, ApiProvider, JournalProvider };
