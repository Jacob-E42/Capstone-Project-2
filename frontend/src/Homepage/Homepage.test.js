import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { AlertProvider, UserProvider } from "../mock";
import Homepage from "./Homepage";

test("Homepage renders without crashing", () => {
	render(
		<MemoryRouter>
			<UserProvider>
				<AlertProvider>
					<Homepage />
				</AlertProvider>
			</UserProvider>
		</MemoryRouter>
	);
});

test("Homepage matches snapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<UserProvider>
				<AlertProvider>
					<Homepage />
				</AlertProvider>
			</UserProvider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
