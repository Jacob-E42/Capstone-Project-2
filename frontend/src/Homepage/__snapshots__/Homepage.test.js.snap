// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p>
      User: testfirst
    </p>
    <h1
      class="mt-5"
    >
      Inkling
    </h1>
    <p
      class="lead"
    >
      A cool slogan here
    </p>
    <div
      class="d-flex justify-content-center"
    >
      <a
        class="mr-3 btn btn-primary"
        href="/signup"
      >
        Sign Up
      </a>
      <a
        class="btn btn-secondary"
        href="/login"
      >
        Log In
      </a>
    </div>
  </div>
</DocumentFragment>
`;
